plugins/pencilblue/controllers/actions/salesforce/logout.js

```diff
@@ -23,10 +23,12 @@ module.exports = function LogOutSFSSOControllerModule(pb) {
             let response = await salesforceStrategyService.logout(this.req);
             try {
                 await new Promise((resolve, reject) => {
-                    pb.session.end(this.session, (err, result) => {
+                    pb.session.end(this.session, async (err, result) => {
                         if (err) {
                             reject(false);
                         } else {
+                            //clear specific values in session for better control
+                            await salesforceStrategyService.clearSpecificValuesInSession(this.session);
```

Comment:
I was doing some tests locally, I couldn't reproduce the logout issue but I get what your trying to prevent by doing these changes. If I'm right, what might cause the error is the failure on the execution of line 23, since the execution of the code within the try-catch is not dependent on the line 23, we can execute  the code in the try-catch first and then execute the code from line 23, that way session data will be deleted first even if the login on SF side is not working.

Reply:
Thanks Sindy. I'll check this locally